src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -181,18 +181,24 @@ public function decode(string $data, string $format, array $context = [])
                 $depth = $headerCount[$i];
                 $arr = &$item;
                 for ($j = 0; $j < $depth; ++$j) {
+                    $headerName = $headers[$i][$j];
+
+                    if ('' === $headerName) {
```

Comment:
I am not sure that we should really allow getting to this place at all. Above the outer `for` loop we check the first line of the CSV to extract headers names. Would it maybe be better to throw a meaningful exception there if empty header names are used?

Reply:
Both solutions look ok to me.

Reply:
Not sure about throwing. It's better to do our best to parse any doc IMHO.